src/twisted/logger/_capture.py

```diff
@@ -0,0 +1,56 @@
+# -*- test-case-name: twisted.logger.test.test_capture -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Context manager for capturing logs.
+"""
+
+from io import StringIO
+
+from attr import attrib, attrs
+
+from twisted.logger import globalLogPublisher, textFileLogObserver
+
+
+
+@attrs(frozen=True, slots=True)
```

Comment:
One other advantage of having this be a `@contextmanager` function is that we don't need to care about frozenness or slotsness :).  Some metaprogramming tools can get a little annoying to use with things that are slotsified and don't have a `__dict__`.